inc/authldap.class.php

```diff
@@ -2659,13 +2678,14 @@ function connect() {
    static function connectToServer($host, $port, $login = "", $password = "",
                                    $use_tls = false, $deref_options = 0,
                                    $tls_certfile = "", $tls_keyfile = "",
-                                   $use_bind = true) {
+                                   $use_bind = true, $timeout = 0) {
 
       $ds = @ldap_connect($host, intval($port));
       if ($ds) {
          @ldap_set_option($ds, LDAP_OPT_PROTOCOL_VERSION, 3);
          @ldap_set_option($ds, LDAP_OPT_REFERRALS, 0);
          @ldap_set_option($ds, LDAP_OPT_DEREF, $deref_options);
+         @ldap_set_option($ds, LDAP_OPT_NETWORK_TIMEOUT, $timeout);
```

Comment:
Maybe this option should not be set if timeout is 0.
```suggestion
         if ($timeout > 0) {
            @ldap_set_option($ds, LDAP_OPT_NETWORK_TIMEOUT, $timeout);
         }
```